lib/vquic/msh3.c

```diff
@@ -121,7 +121,10 @@ CURLcode Curl_quic_connect(struct Curl_easy *data,
     return CURLE_FAILED_INIT;
   }
 
-  qs->conn = MsH3ConnectionOpen(qs->api, conn->host.name, unsecure);
+  qs->conn = MsH3ConnectionOpen(qs->api,
+                                conn->host.name,
+                                (uint16_t)conn->remote_port,
+                                unsecure);
```

Comment:
This will put a hard requirement on a very recent version of the library, that should go into the `--with-msh3` autoconf check to test for this signature no?

Reply:
I think it's very safe if there is a signature checking, but too early to introduce it probably.

Reply:
I think for now it is fine to require a very recent version of msh3 to build. It is early days and I assume it might move a bit going forward. We've worked similar to that with the other h3 libraries. They're all "beta" and this is functionality still labeled "experimental" in curl.